### ISSUE TITLE
DuckDB: Failure to parse `REPLACE` after `EXCLUDE` in `SELECT` clause

### ISSUE THREAD

**Issue by keraion** [sqlfluff/sqlfluff#5554]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Query fails to parse when using the `REPLACE` clause after the `EXCLUDE` clause in a wildcard expression.

### Expected Behaviour

Query should parse successfully.

### Observed Behaviour

```sh
echo "select * exclude (y) replace (3 as x), 6 as z from tabx;" | sqlfluff parse --dialect duckdb -

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    wildcard_expression:
[L:  1, P:  8]      |                        wildcard_identifier:
[L:  1, P:  8]      |                            star:                             '*'
[L:  1, P:  9]      |                    whitespace:                               ' '
[L:  1, P: 10]      |                    keyword:                                  'exclude'
[L:  1, P: 17]      |                    whitespace:                               ' '
[L:  1, P: 18]      |                    bracketed:
[L:  1, P: 18]      |                        start_bracket:                        '('
[L:  1, P: 19]      |                        [META] indent:
[L:  1, P: 19]      |                        column_reference:
[L:  1, P: 19]      |                            naked_identifier:                 'y'
[L:  1, P: 20]      |                        [META] dedent:
[L:  1, P: 20]      |                        end_bracket:                          ')'
[L:  1, P: 21]      |                [META] dedent:
[L:  1, P: 21]      |                whitespace:                                   ' '
[L:  1, P: 22]      |                unparsable:                                   !! Expected: 'Nothing here.'
[L:  1, P: 22]      |                    word:                                     'replace'
[L:  1, P: 29]      |                    whitespace:                               ' '
[L:  1, P: 30]      |                    start_bracket:                            '('
[L:  1, P: 31]      |                    numeric_literal:                          '3'
[L:  1, P: 32]      |                    whitespace:                               ' '
[L:  1, P: 33]      |                    word:                                     'as'
[L:  1, P: 35]      |                    whitespace:                               ' '
[L:  1, P: 36]      |                    word:                                     'x'
[L:  1, P: 37]      |                    end_bracket:                              ')'
[L:  1, P: 38]      |                    comma:                                    ','
[L:  1, P: 39]      |                    whitespace:                               ' '
[L:  1, P: 40]      |                    numeric_literal:                          '6'
[L:  1, P: 41]      |                    whitespace:                               ' '
[L:  1, P: 42]      |                    word:                                     'as'
[L:  1, P: 44]      |                    whitespace:                               ' '
[L:  1, P: 45]      |                    word:                                     'z'
[L:  1, P: 46]      |            whitespace:                                       ' '
[L:  1, P: 47]      |            from_clause:
[L:  1, P: 47]      |                keyword:                                      'from'
[L:  1, P: 51]      |                whitespace:                                   ' '
[L:  1, P: 52]      |                from_expression:
[L:  1, P: 52]      |                    [META] indent:
[L:  1, P: 52]      |                    from_expression_element:
[L:  1, P: 52]      |                        table_expression:
[L:  1, P: 52]      |                            table_reference:
[L:  1, P: 52]      |                                naked_identifier:             'tabx'
[L:  1, P: 56]      |                    [META] dedent:
[L:  1, P: 56]      |    statement_terminator:                                     ';'
[L:  1, P: 57]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:  22 |  PRS | Line 1, Position 22: Found unparsable section: 'replace (3 as x),
                       | 6 as z'
WARNING: Parsing errors found and dialect is set to 'duckdb'. Have you configured your dialect correctly?
```

### How to reproduce

```sh
echo "select * exclude (y) replace (3 as x), 6 as z from tabx;" | sqlfluff parse --dialect duckdb -
```

### Dialect

DuckDB

### Version

Python 3.11.7
sqlfluff, version 3.0.0a4

### Configuration

Defaults

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
